lib/src/asset/id.dart

```diff
@@ -86,11 +89,12 @@ class AssetId implements Comparable<AssetId> {
   AssetId changeExtension(String newExtension) =>
       new AssetId(package, pathos.withoutExtension(path) + newExtension);
 
+  @override
   String toString() => "$package|$path";
 
   /// Serializes this [AssetId] to an object that can be sent across isolates
   /// and passed to [deserialize].
-  serialize() => [package, path];
+  List serialize() => [package, path];
```

Comment:
This was actually intentionally dynamic, but you could mark that explicitly (maybe as `Object` actually?). The format is likely to change in the future and is meant to be treated as opaque.